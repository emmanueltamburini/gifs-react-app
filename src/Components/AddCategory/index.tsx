import { useState } from "react";

interface Props {
  onAdd: (value: string) => void
}

const AddCategory = ({ onAdd }:Props) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({target}: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(target.value);
  }

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    onAdd(inputValue);
    setInputValue('');
  }

  return (
    <form onSubmit={onSubmit}>
      <input
          type="text"
          placeholder="Search gif"
          value={inputValue}
          onChange={onInputChange}
      />
    </form>
  )
}

export default AddCategory;