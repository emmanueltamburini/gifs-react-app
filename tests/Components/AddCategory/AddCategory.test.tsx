import React from "react";
import { fireEvent, render, screen } from "@testing-library/react"
import { AddCategory } from '../../../src/Components/AddCategory/index';
describe('Test in AddCategory', () => {
    test('should show the value of the textbox', () => {
        render(<AddCategory onAdd={jest.fn()} />);
        const input = screen.getByRole<HTMLInputElement>('textbox');

        fireEvent.input(input, {target: {value: 'test'}});

        expect(input.value).toBe('test');
    });

    test('should call onAdd function when the input has a value', () => {
        const inputValue = 'Test';
        const onAdd = jest.fn();

        render(<AddCategory onAdd={onAdd} />);

        const input = screen.getByRole<HTMLInputElement>('textbox');
        const form = screen.getByRole<HTMLFormElement>('form');

        fireEvent.input(input, {target: {value: inputValue}});
        fireEvent.submit(form);
        
        expect(input.value).toBe('');
        expect(onAdd).toHaveBeenCalled();
        expect(onAdd).toHaveBeenCalledTimes(1);
        expect(onAdd).toHaveBeenCalledWith(inputValue);
    });

    test('should call onAdd function when the input has no value', () => {
        const onAdd = jest.fn();

        render(<AddCategory onAdd={onAdd} />);

        const input = screen.getByRole<HTMLInputElement>('textbox');
        const form = screen.getByRole<HTMLFormElement>('form');

        fireEvent.submit(form);
        
        expect(input.value).toBe('');
        expect(onAdd).toHaveBeenCalled();
        expect(onAdd).toHaveBeenCalledWith('');
    });
})