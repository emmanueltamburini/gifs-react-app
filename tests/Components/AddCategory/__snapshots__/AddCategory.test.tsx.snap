// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory should be equal to snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
